src/Admin/BaseMediaAdmin.php

```diff
@@ -118,7 +122,13 @@ public function alterNewInstance(object $object): void
             if (null !== $this->categoryManager && $categoryId = $this->getPersistentParameter('category')) {
                 $category = $this->categoryManager->find($categoryId);
 
-                if ($category && $category->getContext()->getId() === $context) {
+                if (!$category) {
+                    return;
+                }
+
+                $categoryContext = $category->getContext();
+
+                if ($categoryContext && method_exists($categoryContext, 'getId') && $categoryContext->getId() === $context) {
```

Comment:
Can't https://github.com/sonata-project/SonataMediaBundle/issues/1978#issuecomment-861252460 could solve all the method_exists checks ?

Reply:
It depends on how Classification has implemented their interfaces, at some point we will have to decide what to do here, but there are bundles that contains the getId methods on the interface and other not.

Reply:
Solved